pandas/_libs/tslibs/offsets.pyx

```diff
@@ -5359,6 +5371,11 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
+                if name in deprec_to_valid_alias:
+                    raise ValueError(INVALID_FREQ_ERR_MSG.format(
```

Comment:
Instead, could you make `INVALID_FREQ_ERR_MSG` a function like:

```python
def raise_invalid_freq(freq: str, extra_message: str | None = None): -> None:
    msg = f"Invalid frequency: {freq}"
    if extra_message is not None:
        msg += extra_message
    if name in deprc_to_valid_alias:
        msg += "Did you mean {deprec_to_valid_alias[name]}?"
    raise ValueError(msg)
```

Can call this function wherever `INVALID_FREQ_ERR_MSG`. It appears this variable is used in multiple places so it would be nice to consistently provide this hint